packages/api/README.md

```diff
@@ -0,0 +1,160 @@
+# API
+
+The nft.storage public API.
+
+## Getting Started
+
+### Local env vars
+
+Inside the `/packages/api` folder create a file called `.env.local` with the following content.
+
+```ini
+SENTRY_TOKEN=<sentry user auth token>
+SENTRY_UPLOAD=false # toggle for sentry source/sourcemaps upload (capture will still work)
+```
+
+Production vars should be set in Github Actions secrets.
+
+### Cloudflare Workers CLI
+
+```bash
+yarn global add @cloudflare/wrangler
+wrangler login
+# when using personal accounts you may need to manually change the `account_id` inside `wrangler.toml`
+```
+
+### Magic.link account
+
+Go to [magic.link](https://magic.link) and create an account. Create two applications one for dev and another for production. In the "settings" of each application you will find the secrets needed to complete the initial setup.
+
+### IPFS Cluster
+
+The nft.storage site talks to IPFS Cluster. You need to run a cluster locally and make it accessible from the internet for development.
+
+Follow the quickstart guide to get an IPFS Cluster up and running: https://cluster.ipfs.io/documentation/quickstart/
+
+Expose the IPFS Proxy by opening the `docker-compose.yml` file and add:
+
+```yaml
+CLUSTER_IPFSPROXY_NODEMULTIADDRESS: /dns4/ipfs0/tcp/5001
+CLUSTER_IPFSPROXY_LISTENMULTIADDRESS: /ip4/0.0.0.0/tcp/9095
+```
+
+...to the `custer0` _environment_ and add:
+
+```yaml
+- '127.0.0.1:9095:9095'
+```
+
+...to the `cluster0` _ports_. Then restart for the changes to take effect.
+
+Install [localtunnel](https://localtunnel.me/) and expose the IPFS Cluster HTTP API and IPFS Proxy API (replacing "USER" with your name):
+
+```
+npm install -g localtunnel
+lt --port 9094 --subdomain USER-cluster-api-nft-storage
+lt --port 9095 --subdomain USER-ipfs-proxy-api-nft-storage
+```
+
+These two URLs should be used for `CLUSTER_API_URL` and `CLUSTER_IPFS_PROXY_API_URL` in the `wrangler.toml` (see below).
+
+There is an npm script you can use to quickly establish these tunnels during development:
+
+```sh
+yarn lt
+```
+
+### Cloudflare Workers initial setup:
+
+> This only needs to be run once when setting up from scratch.
+
+#### Development Setup
+
+Open `wrangler.toml` and add an env for yourself (replacing "USER" with your name and "CF_ACCOUNT" with your Cloudflare account):
+
+```toml
+[env.USER]
+type = "webpack"
+name = "nft-storage-USER"
+account_id = "CF_ACCOUNT"
+workers_dev = true
+route = ""
+zone_id = ""
+vars = { ENV = "dev", DEBUG = "*", CLUSTER_API_URL = "", CLUSTER_IPFS_PROXY_API_URL = "" }
+kv_namespaces = []
+```
+
+Additionally, fill in the `CLUSTER_API_URL` and `CLUSTER_IPFS_PROXY_API_URL` with the localtunnel URLs you obtained when setting up the IPFS Cluster.
```

Comment:
Please add an example below, given that we need to append `/api/v0/` to the `CLUSTER_IPFS_PROXY_API_URL`